packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js

```diff
@@ -74,27 +74,44 @@ const withLocalNPMRepo = (func) => {
         .then(
             () =>
                 new Promise((resolve) => {
-                    const logStream = fs.createWriteStream(logFileName, {flags: 'a'})
                     console.log('Starting up local NPM repository')
 
-                    child = sh.exec(`${verdaccio} --config config.yaml`, {
+                    child = cp.exec(`${verdaccio} --config config.yaml`, {
                         cwd: verdaccioConfigDir,
-                        async: true,
-                        fatal: true,
-                        silent: true
-                    })
-
-                    child.stdout.on('data', (data) => {
-                        if (data.includes('http address')) {
-                            // Verdaccio is running once it logs the HTTP address. Configure
-                            // NPM to use the local repo, through env vars.
-                            process.env['npm_config_registry'] = 'http://localhost:4873/'
-                            resolve()
+                        stdio: 'inherit',
+                        env: {
+                            ...process.env,
+                            OPENCOLLECTIVE_HIDE: 'true',
+                            DISABLE_OPENCOLLECTIVE: 'true',
+                            OPEN_SOURCE_CONTRIBUTOR: 'true'
                         }
                     })
 
-                    child.stdout.pipe(logStream)
-                    child.stderr.pipe(logStream)
+                    const checkTime = 1000
+
+                    const waitForLogFileExists = () => {
+                        setTimeout(() => {
+                            fs.readFile(logFileName, (err) => {
+                                if (err) {
+                                    waitForLogFileExists()
+                                } else {
+                                    const readStream = fs.createReadStream(logFileName)
+
+                                    readStream.on('data', (data) => {
+                                        if (data.includes('http address')) {
+                                            // Verdaccio is running once it logs the HTTP address. Configure
+                                            // NPM to use the local repo, through env vars.
+                                            process.env['npm_config_registry'] =
+                                                'http://localhost:4873/'
+                                            resolve()
+                                        }
+                                    })
+                                }
+                            })
+                        }, checkTime)
+                    }
```

Comment:
Can you explain why you're waiting for this log file, please? The comment in the code says _"Verdaccio is running once it logs the HTTP address"_. Is that now incorrect?

Reply:
Yes, the comment is still valid. 
The Verdaccio configuration has been updated to output all the Verdaccio's logs to a verdaccio.log file using the native Verdaccio functionality.

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/986e5da75bbd1fd19e4fb03d550a500470ec9db3/packages/pwa-kit-create-app/local-npm-repo/config.yaml#L32-L33

https://verdaccio.org/docs/logger/

The `create-mobify-app-dev.js` script now monitors the verdaccio.log file content and waits until Verdaccio is ready before continuing with the rest of the promise chain.

The previous `create-mobify-app-dev.js` script was not working on a Windows Circle CI machine (I believe we never ran the script `create-mobify-app-dev.js` on Windows before ). The process hung without any output for more than 10 min.

Potentially the problem was related to the terminal output piping between the different processes run in the promise chain. The Verdaccion logs were overwritten and lost by the following processes output in the promises, ending in a situation where the script didn't output anything on Windows, nor the expected zero/non-zero exit by Circle CI.

Reply:
Sorry, I meant, "can you explain this with a comment in the code?" The existing comment is now incorrect.

Reply:
Oh, I get it. You've redirected the output to a log file so now you're looking at that instead. Sounds good to me – I think, on reflection, the comment still makes sense.